MekHQ/src/mekhq/campaign/mission/ScenarioDeploymentLimit.java

```diff
@@ -0,0 +1,469 @@
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Determines whether a given unit is part of a Force
+     * @param unitId - the id of a Person
+     * @param force - a Force
+     * @param c - a pointer to the campaign
+     * @return a boolean that evaluates to true if the unit identified by unitId is part of the force
+     */
+    public boolean isUnitInForce(UUID unitId, Force force, Campaign c) {
+        Vector<UUID> unitIds = force.getAllUnits(true);
+        for (UUID id : unitIds) {
+            if (id.equals(unitId)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Returns a String giving a description of required units for graphical display
+     * @param c - a pointer to the campaign
+     * @return a String that is a comma separated list of required unit names
+     */
+    public String getRequiredUnitDesc(Campaign c) {
+        ArrayList<String> unitNames = new ArrayList<String>();
+        for (UUID unitId: requiredUnits) {
+            Unit u = c.getUnit(unitId);
+            if (null != u) {
+                unitNames.add(u.getName());
+            }
+        }
+        return String.join(", ", unitNames);
+    }
+    //endregion Required unit methods
+
+    //region File I/O
+    @Override
+    public void writeToXml(PrintWriter pw1, int indent) {
+        pw1.println(MekHqXmlUtil.indentStr(indent) + "<scenarioDeploymentLimit>");
+        pw1.println(MekHqXmlUtil.indentStr(indent+1)
+                +"<quantityLimit>"
+                +quantityLimit
+                +"</quantityLimit>");
+        pw1.println(MekHqXmlUtil.indentStr(indent+1)
+                +"<quantityType>"
+                +quantityType.name()
+                +"</quantityType>");
+        pw1.println(MekHqXmlUtil.indentStr(indent+1)
+                +"<countType>"
+                +countType.name()
+                +"</countType>");
+        for(UUID id : requiredPersonnel) {
+            pw1.println(MekHqXmlUtil.indentStr(indent+1)
+                    +"<requiredPersonnel>"
+                    +id
+                    +"</requiredPersonnel>");
+        }
+        for(UUID id : requiredUnits) {
+            pw1.println(MekHqXmlUtil.indentStr(indent+1)
+                    +"<requiredUnit>"
+                    +id
+                    +"</requiredUnit>");
+        }
+        if (!allowedUnitTypes.isEmpty()) {
+            pw1.println(MekHqXmlUtil.indentStr(indent+1) + "<allowedUnitTypes>");
+            for(int type : allowedUnitTypes) {
+                pw1.println(MekHqXmlUtil.indentStr(indent+2)
+                        +"<allowedUnitType>"
+                        +type
+                        +"</allowedUnitType>");
+            }
+            pw1.println(MekHqXmlUtil.indentStr(indent+1) + "</allowedUnitTypes>");
+        }
+        pw1.println(MekHqXmlUtil.indentStr(indent) + "</scenarioDeploymentLimit>");
```

Comment:
This needs to be converted to the latest MekHqXmlUtil methods, plus the spacing needs fixing.

Reply:
Can you give me an example of a class using the latest methods?